app/models/concerns/audit_extensions.rb

```diff
@@ -33,9 +34,26 @@ def ensure_taxonomies_not_escalated
   private
 
   def log_audit
-    Foreman::Logging.with_fields(self.audited_changes) do
-      Foreman::Logging.logger('audit').info { "#{self.action} event for #{self.auditable_type} with id #{self.auditable_id}" }
+    telemetry_increment_counter(:audit_records_created, 1, type: self.auditable_type)
+    return unless (self.audited_changes && Foreman::Logging.logger('audit').info?)
+    self.audited_changes.each_pair do |column_name, old_and_new|
+      audited_fields = {
+        audit_action: self.action,
+        audit_type: self.auditable_type,
+        audit_id: self.auditable_id,
+        audit_column: column_name
+      }
+      if old_and_new.is_a?(Array)
+        audited_fields[:audit_field_old] = old_and_new[0]
+        audited_fields[:audit_field_new] = old_and_new[1]
+      else
+        audited_fields[:audit_field] = old_and_new
+      end
+      Foreman::Logging.with_fields(audited_fields) do
+        Foreman::Logging.logger('audit').info "#{self.auditable_type} (#{self.auditable_id}) #{self.action} event on #{column_name}"
```

Comment:
This results in lines like "Subnet (12) create event on network", it's great I can grep the line so I know subnet 12's network has been updated but I don't see to what, unless I'm parsing journal structured data. Could you add the `old_and_new` info to this log please?